balancer/weightedroundrobin/weightedroundrobin.go

```diff
@@ -19,11 +19,47 @@
 // Package weightedroundrobin defines a weighted roundrobin balancer.
 package weightedroundrobin
 
+import (
+	"google.golang.org/grpc/resolver"
+)
+
 // Name is the name of weighted_round_robin balancer.
 const Name = "weighted_round_robin"
 
-// AddrInfo will be stored inside Address metadata in order to use weighted roundrobin
-// balancer.
+// attributeKey is the type used as the key to store AddrInfo in the Attributes
+// field of resolver.Address.
+type attributeKey struct{}
+
+// AddrInfo will be stored inside Address metadata in order to use weighted
+// roundrobin balancer.
 type AddrInfo struct {
 	Weight uint32
 }
+
+// SetAddrInfo returns a copy of addr in which the Attributes field is updated
+// with addrInfo.
+//
+// This is an EXPERIMENTAL API.
+func SetAddrInfo(addrInfo AddrInfo, addr resolver.Address) resolver.Address {
+	newAddr := addr
```

Comment:
Since this is passed by value, there's no need to make this copy here.

Reply:
Done.